### PR TITLE
Calculate L1phase when statistics reached value

### PR DESCRIPTION
**Pull request by peressounko**

### CODE REVIEW

File: Detectors/PHOS/calib/src/PHOSL1phaseCalibDevice.cxx [hunk 1]
```diff
@@ -41,19 +41,46 @@ void PHOSL1phaseCalibDevice::run(o2::framework::ProcessingContext& pc)
   auto cellTR = pc.inputs().get<gsl::span<TriggerRecord>>("cellTR");
   LOG(detail) << "Processing TF with " << cells.size() << " cells and " << cellTR.size() << " TrigRecords";
   mCalibrator->process(tfcounter, cells, cellTR);
+
+  ++mNprocessed;
+  // If sufficient statistics was collected, try to calculate L1phases
+  // If statistics reached N*thresholds try to re-calculate L1phases
+  // if they changed wrt previous, replace ccdb object, otherwise do nothing
+  if (mNprocessed % minStatisticsForCalib == 0) {
+    minStatisticsForCalib *= 2;
```
Comment (shahor02) [inline Detectors/PHOS/calib/src/PHOSL1phaseCalibDevice.cxx:50]: Note that you update the calibration not at `N*trhesholds` but at `2^(N-1) * thresholds`. Why do you need to `minStatisticsForCalib *= 2;` after each update?

File: Detectors/PHOS/calib/src/PHOSL1phaseCalibDevice.cxx [hunk 2]
```diff
@@ -41,19 +41,46 @@ void PHOSL1phaseCalibDevice::run(o2::framework::ProcessingContext& pc)
...
+    mCalibrator->checkSlotsToFinalize(o2::calibration::INFINITE_TF);
+    mCalibrator->endOfStream();
```
Comment (shahor02) [inline Detectors/PHOS/calib/src/PHOSL1phaseCalibDevice.cxx:52]: just to be sure: `mCalibrator->endOfStream()` does not really mean endOfStream, but just to finalize the calibration for the next  portion of TFs, right?

File: Detectors/PHOS/calib/src/PHOSL1phaseCalibDevice.cxx (hunk 1 above)
Comment (peressounko) [reply Detectors/PHOS/calib/src/PHOSL1phaseCalibDevice.cxx:50]: L1 object is unique for whole run. Normally it should be calculated from minimal threshold. If in the run we have noisy channels, I think it is better to collect considerably more statistics before next try. This should reduce the number of overlapped objects in ccdb.

File: Detectors/PHOS/calib/src/PHOSL1phaseCalibDevice.cxx (hunk 2 above)
Comment (peressounko) [reply Detectors/PHOS/calib/src/PHOSL1phaseCalibDevice.cxx:52]: Yes, mCalibrator::endOfStream() calculates calibrations but does not reset anything.

File: Detectors/PHOS/calib/src/PHOSL1phaseCalibDevice.cxx (hunk 1 above)
Comment (shahor02) [reply Detectors/PHOS/calib/src/PHOSL1phaseCalibDevice.cxx:50]: ok, thanks, though then the comment about `N*thresholds`  is somewhat misleading...

**Review by shahor02 — APPROVED** [review]

**Pull request merged**